### PR TITLE
fix(appeals/api): allow unaccompanied site visits to be saved without a visit time

### PR DESCRIPTION
**Pull request by shepda**
## Describe your changes

Updated the Appeals Site Visit POST/PATCH API endpoints to allow unaccompanied site visits to be saved without a visit time.

## Issue ticket number and link

https://pins-ds.atlassian.net/browse/BOAT-488

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: appeals/api/src/server/utils/compare-strings.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/**
+ * @param {string} stringA
+ * @param {string} stringB
+ * @returns {boolean}
+ */
+const compareStrings = (stringA, stringB) =>
```
Comment (fabfab-kc) [inline appeals/api/src/server/utils/compare-strings.js:6]: Just a suggestion, wouldn't be clearer with a function name that indicates testing for equality?

**Review by fabfab-kc — APPROVED** [review]
LGTM! Just a suggestion on the naming of the compare util...

File: appeals/api/src/server/utils/compare-strings.js (hunk 1 above)
Comment (shepda) [reply appeals/api/src/server/utils/compare-strings.js]: Yes, good point, I've updated that to `checkStringsMatch`

**Review by fabfab-kc — APPROVED** [review]
great stuff!

**Pull request merged**